### PR TITLE
More prep for native ESM

### PR DESCRIPTION
**Pull request by nwalters512**
This PR makes a few more changes to get us ready for native ESM:

- Avoids using named imports from `lodash`, which Node's ESM compatibility layer for CJS modules doesn't support.
- Replaces `const debug = require('debug')(...)` with `import debugfn from 'debug'`.
- Updates `tools/cjs-to-esm-candidates.mjs` to support the above pattern of immediately calling a required function.
- For `debug`, use hardcoded names instead of dynamically selecting them based on the filename
- Run `tools/cjs-to-esm-candidates.mjs` in CI to prevent regressions while we work to move fully to native ESM

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -353,6 +353,9 @@ jobs:
       - name: Check for duplicate Node dependencies
         run: yarn dedupe --check
 
+      - name: Check for CJS requires which could be ESM imports
+        run: node ./tools/cjs-to-esm-candidates.mjs --check
+
```
Comment (nwalters512) [inline .github/workflows/main.yml:358]: Thoughts on running this check in CI? This has the potential to break CI on WIP PRs, but I think it's worth it to force folks to match our expectations before they can merge.

Comment (jonatanschroeder) [reply .github/workflows/main.yml:358]: I'm in favour.

**Review by jonatanschroeder — APPROVED** [review]

File: tools/cjs-to-esm-candidates.mjs
Comment (jonatanschroeder) [inline tools/cjs-to-esm-candidates.mjs:1]: I can't comment there directly, but I believe some of the modules in lines 63-67 can be removed from this list already, right?

Comment (nwalters512) [reply tools/cjs-to-esm-candidates.mjs:1]: Good call! Will change.

Comment (nwalters512) [reply tools/cjs-to-esm-candidates.mjs:1]: Done in 484275b.

**Pull request merged**